clang/test/CIR/CodeGen/complex.c

```diff
@@ -0,0 +1,130 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir -o - %s | FileCheck %s --check-prefixes=C,CHECK
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -x c++ -fclangir -emit-cir -o - %s | FileCheck %s --check-prefixes=CPP,CHECK
+
+#ifdef __cplusplus__
+#define VOID
+#else
+#define VOID void
+#endif
+
+double _Complex c, c2;
+int _Complex ci, ci2;
+
+volatile double _Complex vc, vc2;
+volatile int _Complex vci, vci2;
+
+void list_init(VOID) {
+  double _Complex c1 = {1.0, 2.0};
+  int _Complex c2 = {1, 2};
+}
+
+//          C: cir.func @list_init()
+//        CPP: cir.func @_Z9list_initv()
+//      CHECK:   %[[#REAL:]] = cir.const #cir.fp<1.000000e+00> : !cir.double
+// CHECK-NEXT:   %[[#IMAG:]] = cir.const #cir.fp<2.000000e+00> : !cir.double
+// CHECK-NEXT:   %{{.+}} = cir.complex.create %[[#REAL]], %[[#IMAG]] : !cir.double -> !cir.complex<!cir.double>
```

Comment:
```
  %0 = cir.const #cir.fp<1.000000e+00> : !cir.double
  %1 = cir.const #cir.fp<2.000000e+00> : !cir.double
  %2 = cir.complex.create %0, %1 : !cir.complex<!cir.double>
  cir.store %2, ... : !cir.ptr<!cir.complex<!cir.double>>
```

I prefer keeping the `#cir.complex` to be used directly with `cir.const`, it composes well with the rest of CIR consts. By applying that change, `cir.complex.create` becomes only useful for "merging" two non-constants fp/ints in order to build a `!cir.complex`. But this brings the question if this is really useful, two possible approaches:

1. We remove it in favor of "get_real/get_imag + store to each", which reuses existing ops. A single store would still be used for when `cir.const` creates the complex type, which also brings some inconsistency.
2. We keep it. Useful to "join" two non-constants into a complex type so that we only have one final store instead of two, theoretically easier to track on analysis. The downside is having an extra op to build and later lower.

Perhaps we should do (2) because it's easier to translate to MLIR's complex dialect?